src/views/data/ControlBar.js

```diff
@@ -1,89 +1,79 @@
-import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
-import _ from 'lodash';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
 import timeout from 'util/timeout';
+import _ from 'lodash';
 
 import {
     getTableSearch,
-    setTableSearch
+    setTableSearch,
+    getTableFilter,
 } from 'core/reducer';
 
-import {useModals} from 'util/ModalProvider';
+import API from 'core/api';
+
+import { useModals } from 'util/ModalProvider';
 import FilterDataModal from './FilterDataModal';
-import {message, Input, Button, Menu, Dropdown, Icon} from 'antd';
+import { message, Input, Button, Menu, Dropdown, Icon } from 'antd';
 
 import './ControlBar.scss';
 
 // TODO: Pass data here...
-const exportCsv = async (data) => {
+const exportCSV = async (filter, search, dataIsFiltered) => {
     const hide = message.loading('Preparing export...');
-
-    // generate export
-    const csvString = `
-col1,col2
-val11,val12
-val21,val22`;
-    await timeout(1000);
-
-    const element = document.createElement('a');
-    element.href = URL.createObjectURL(
-        new Blob([csvString]),
-        { type: 'application/csv' }
-    );
-    element.download = 'export.csv';
-    element.click();
-
+    // go to new page with url as query
+    if (dataIsFiltered) {
+        const tableQuery = { filter: filter, search: search }
+        window.open(await API.exportFilteredCSV(tableQuery), '_blank');
+    } else {
+        window.open(await API.exportAllCSV(), '_blank');
+    }
     hide();
     message.success('Export ready. Downloading...');
 };
 
 const ControlBar = useModals(connect(
     (state) => ({
-        search: getTableSearch(state)
+        search: getTableSearch(state),
+        filter: getTableFilter(state),
     }),
     (dispatch) => bindActionCreators({
         setSearch: setTableSearch
     }, dispatch)
-)(({ search, setSearch, openModal }) => {
-    const debouncedSearch = _.debounce(setSearch, 500);
-    const onSearchChange = ({ target: { value } }) => debouncedSearch(value);
-
-    return (
-        <div className='control-bar'>
-            <Input
-                placeholder='Quick search'
-                className='quick-search'
-                allowClear={true}
-                defaultValue={search}
-                onChange={onSearchChange}
-            />
+)(({ filter, search, setSearch, openModal }) => (
+    <div className='control-bar'>
+        <Input
+            placeholder='Quick search'
+            className='quick-search'
+            allowClear={true}
+            value={search}
+            onChange={({ target: { value } }) => setSearch(value)}
+        />
 
+        <Button
+            type='primary'
+            icon='tool'
+            onClick={() => openModal(<FilterDataModal />)}
+        >
+            Filter...
+        </Button>
+        <Dropdown overlay={
+            <Menu
+                onClick={({ key }) => exportCSV(filter, search, key === 'filter')}
```

Comment:
If you change the `exportCSV` function as I mentioned above, you could just change `onClick` to:
```JavaScript
onClick={({ key}) => key === filter ? exportCSV(filter, search) : exportCSV()}
````

Reply:
Changed. good call chief